File: sdk/csharp/libraries/microsoft.bot.builder.solutions/Skills/Models/EnhancedBotFrameworkSkill.cs
﻿using Microsoft.Bot.Builder.Skills;

namespace Microsoft.Bot.Builder.Solutions.Skills.Models
{
    // Enhanced version of BotFrameworkSkill that adds additional properties commonly needed by a Skills VA
    public class EnhancedBotFrameworkSkill : BotFrameworkSkill
    {
        // Summary:
        //     Gets or sets the Name of the skill.
        public string Name { get; set; }

        // Summary:
        //     Gets or sets the Description of the skill.
        public string Description { get; set; }
    }
}

File: sdk/typescript/libraries/botbuilder-solutions/src/skills/enhancedBotFrameworkSkill.ts
/**
 * Copyright(c) Microsoft Corporation.All rights reserved.
 * Licensed under the MIT License.
 */

import { BotFrameworkSkill } from 'botbuilder';

/**
* Enhanced version of BotFrameworkSkill that adds additional properties commonly needed by a Skills VA
*/
export interface IEnhancedBotFrameworkSkill extends BotFrameworkSkill {

    /**
     * Gets or sets the Name of the skill.
     */
    name: string;

    /**
     * Gets or sets the Description of the skill.
     */
    description: string;
}
